### PR TITLE
Update api client to jest - Closes #4678

### LINKED ISSUE #4678: Convert lisk-api-client mocha test to jest
### Description

Convert the tests in `lisk-api-client` library from `mocha` to `jest`.

### Motivation

Consistent use of `jest` testing.

### Acceptance Criteria

`jest`

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?
It was using mocha

### How did I solve it?
- Update the file names
- Update the syntax

### How to manually test it?
- All test runs successfully
- Coverage is taken

### Review checklist

- [ ] The PR resolves #4678 
- [ ] All new code is covered with unit tests
- [ ] Relevant integration / functional tests are added
- [ ] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [ ] Documentation has been added/updated

### CODE REVIEW

File: elements/lisk-api-client/test/api_client.spec.ts [hunk 1]
```diff
@@ -209,56 +194,45 @@ describe('APIClient module', () => {
...
+				return expect(apiClient.headers).not.toHaveProperty('User-Agent');
 			});
 		});
 
 		describe('nodes', () => {
 			it('should have nodes supplied to constructor', () => {
-				return expect(apiClient)
-					.to.have.property('nodes')
-					.and.equal(defaultNodes);
+				return expect(apiClient.nodes).toBe(defaultNodes);
```
Comment (valamidev) [inline elements/lisk-api-client/test/api_client.spec.ts]: expect(apiClient).toHaveProperty('nodes', defaultNodes);

Comment (shuse2) [reply elements/lisk-api-client/test/api_client.spec.ts]: I think they are essentially the same?

Comment (valamidev) [reply elements/lisk-api-client/test/api_client.spec.ts]: Sure, second is more readable as it refer that it is an object.

Comment (shuse2) [reply elements/lisk-api-client/test/api_client.spec.ts]: Will try to use it in other PRs

**Pull request merged**